styx-common/src/main/java/com/spotify/styx/state/RunState.java

```diff
@@ -172,8 +172,20 @@ public RunState created(WorkflowInstance workflowInstance, String executionId, S
     }
 
     @Override
+    public RunState dequeue(WorkflowInstance workflowInstance) {
+      switch (state()) {
+        case QUEUED:
+          return state(PREPARE);
+
+        default:
+          throw illegalTransition("dequeue");
+      }
+    }
+
+    @Override
     public RunState submit(WorkflowInstance workflowInstance, ExecutionDescription executionDescription) {
       switch (state()) {
+        case QUEUED: // for backwards compatibility
```

Comment:
Since we added the QUEUED state after deprecating `created`, we should not have in storage a `created` transition that was executed on top of a state machine that includes the QUEUED state. Do we really need to allow for that transition in RunState?